conans/test/functional/command/test_command_test.py

```diff
@@ -30,3 +33,10 @@ def test_conan_test(self):
         subfolder = "Release" if platform.system() != "Windows" else ""
         assert os.path.exists(os.path.join(client.current_folder, "test_package",
                                            "build", subfolder, "generators", "conaninfo.txt"))
+
+    def test_whateverrr(self):
+        tc = TestClient()
+        tc.run("new hello/1.0 -m=cmake_lib")
+        test_root_folder_output = temp_folder()
+        tc.run(f"create . hello/1.0@ -c user.test_build_folder='{test_root_folder_output}'")
```

Comment:
Not enough testing. The functionality should already create new different ``test_package/build/gcc_10_sha1`` folders for different configs, even if no ``user.conf`` is provided.

Then, for the other functionality of relocating the whole would be probably a  ``core.xxxx`` one.